content/en/docs/concepts/services-networking/_index.md

```diff
@@ -7,44 +7,195 @@ description: >
+{{< /note >}}
+
+### Containers on the same pod
+
+Inter-container connectivity on the same Pod makes use of pod IPs, network namespaces and localhost networking.
+
+Pods run in their own [network namespace](https://man7.org/linux/man-pages/man7/network_namespaces.7.html). All the containers in a Pod share this network namespace. (Network namespaces are not the same as the Kubernetes {{< glossary_tooltip text="namespace" term_id="namespace" >}} concept).
+
+Figure 2 illustrates an example of localhost communications between two containers on the same Pod. 
+
+{{< figure src="/docs/images/k8s-localhost-02.drawio.svg" alt="diagram of K8s local networking for containers in a single Pod" class="diagram-large" caption="Figure 2. Example of container communications within the same Pod using localhost:port number" >}}
+
+Your Pod 1 containers share a network namespace, IP address and MAC address. However, you do need to configure separate port numbers for each container.
+
+### Pods on the same node
+
+Inter-pod connectivity on the same node utilize pod IPs, Pod and root network namespaces, [veth links](https://man7.org/linux/man-pages/man4/veth.4.html) and a [virtual L2 bridge](https://en.wikipedia.org/wiki/Data_link_layer).
+
+Pods operate in their own network namespace and have their own IP and MAC addresses. You can think of this as multiple distinct IP computers (pods), that wish to talk with each other through a L2 bridge. Inter-pod networking on the same node is no different.
+
+This example of inter-pod networking on the same node requires:
+
+* Root network namespace for the node. 
+
+* Veth links enabling communication between the Pod network namespace and root network namespace. 
+
+* L2 or L3 networking between the Pods. This example uses a virtual L2 bridge. You might use a different approach supported by your network plugin.
+
+Figure 3 illustrates an example of two Pods talking to each other on the same node. 
+
+{{< figure src="/docs/images/k8net-PodSameHost04.drawio.svg" alt="diagram of 2 Pods on the same node using an L2 bridge" class="diagram-large" caption="Figure 3. Example of Pod 1 and Pod 2 on the same node communicating through a L2 bridge" >}}
+
+### Pods on different nodes
+
+#### Physical network
+
+Your Kubernetes deployment might place Pods on a network of nodes. With this topology, you can distribute your workloads across many Pods to increase scale and resiliency. Inter-pod networking between distinct nodes becomes a crucial function for your cluster network. 
+
+In all scenarios involving inter-node Pod networking (physical or virtual), you will have a physical network in place. This provides you with communications between different network resources needed for K8s and non-K8s application support, APIs, metrics collection, operations and troubleshooting. 
+
+For a K8s-specific example, you could have a process call the control plane API server to issue an HTTP request to a node kubelet. 
```

Comment:
I can not fully understand this sentence sorry?

Reply:
the way this can work is the kubelet register itsefl with the apiserver creating the corresponding node object that represent itself.
It can also happen that the apiserver connects to the kubelet to perform some operations like "kubectl exec" per example

Reply:
You might only be configuring a virtual network; for example, in the cloud, where the underlying physical network is abstracted away.